src/settings.rs

```
@@ -238,35 +241,42 @@ impl ContextSettingInternalExt for Context {
         setting_type: TypeId,
         setting_id: usize,
         q: T,
-    ) -> Option<PersonId> {
-        self.get_setting_members_internal(setting_type, setting_id)
-            .and_then(|members| {
-                if members.len() == 1 {
-                    return None;
-                }
-                let member_iter = members.iter().filter(|&x| *x != person_id);
-
-                let mut contacts = vec![];
-                if q.get_query().is_empty() {
-                    // If the query is empty we push members directly to the vector
-                    for contact in member_iter {
-                        contacts.push(*contact);
-                    }
-                } else {
-                    // If the query is not empty, we match setting members to the query
-                    for contact in member_iter {
-                        if self.match_person(*contact, q) {
-                            contacts.push(*contact);
-                        }
-                    }
-                }
+    ) -> Result<Option<PersonId>, IxaError> {
+        let members = self
+            .get_setting_members_internal(setting_type, setting_id)
+            .unwrap();
```

I don't think this should be an unwrap. If it's an unwrap, when `get_setting_members_internal` returns None, this code panics rather than returning `Ok(None)`. I think you can keep the `.and_then` structure and just add a check that confirms that `members` contains the person_id and if it doesn't return an error.

---

rather than worrying about the and_then implementation should we throw an error if the let members = self.get_setting_members_internal(setting_type, setting_id); is none when calling the get get_contact_internal function. It seems like something would be wrong if thats the case. It is similar to if the person_Id is not in the membership list.